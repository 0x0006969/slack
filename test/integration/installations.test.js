const helper = require('.');
const installationCreated = require('../fixtures/webhooks/installation.created.json');
const installationDeleted = require('../fixtures/webhooks/installation.deleted.json');

describe('Integration: tracking GitHub installations', () => {
  test('installation created records', async () => {
    const { robot } = helper;
    const { Installation } = robot.models;

    await robot.receive({
      event: 'installation',
      payload: installationCreated,
    });

    let installation = await Installation.findOne({
      where: { githubId: installationCreated.installation.id } });

    console.log('WTF?', await Installation.findAll());

    expect(installation).toBeTruthy();

    await robot.receive({
      event: 'installation',
      payload: installationDeleted,
    });

    installation = await Installation.findOne({
      where: { githubId: installationCreated.installation.id } });

    expect(installation).toBe(null);
  });
});
