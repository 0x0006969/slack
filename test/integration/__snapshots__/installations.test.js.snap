// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration: tracking GitHub installations deleting installation cascades to delete all subscriptions related to that installation 1`] = `[MockFunction cache.get]`;

exports[`Integration: tracking GitHub installations deleting installation cascades to delete all subscriptions related to that installation 2`] = `[MockFunction cache.set]`;

exports[`Integration: tracking GitHub installations deleting installation cascades to delete all subscriptions related to that installation 3`] = `[MockFunction cache.fetch]`;

exports[`Integration: tracking GitHub installations installation created records 1`] = `[MockFunction cache.get]`;

exports[`Integration: tracking GitHub installations installation created records 2`] = `[MockFunction cache.set]`;

exports[`Integration: tracking GitHub installations installation created records 3`] = `[MockFunction cache.fetch]`;
